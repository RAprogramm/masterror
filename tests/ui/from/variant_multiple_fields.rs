// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::Error;

#[derive(Debug, Error)]
enum BadEnum {
    #[error("{source:?} - {extra:?}")]
    WithExtra {
        #[from]
        source: DummyError,
        #[backtrace]
        trace: Option<std::backtrace::Backtrace>,
        extra: DummyError
    }
}

#[derive(Debug, Error)]
#[error("dummy")]
struct DummyError;

fn main() {}
