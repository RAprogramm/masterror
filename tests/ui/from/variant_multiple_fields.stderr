error: deriving From requires no fields other than source and backtrace
  --> tests/ui/from/variant_multiple_fields.rs:11:9
   |
11 |         #[from]
   |         ^^^^^^^
